File: library/src/main/java/com/github/ykrank/androidtools/binding/LibRecyclerViewBindingAdapter.java
package com.github.ykrank.androidtools.binding;

import androidx.recyclerview.widget.RecyclerView;

import com.github.ykrank.androidtools.ui.adapter.LibBaseRecyclerViewAdapter;


public final class LibRecyclerViewBindingAdapter {

    private LibRecyclerViewBindingAdapter() {
    }

    public static void setHasProgress(RecyclerView recyclerView, Boolean oldIsLoadingFirstTime, Boolean newIsLoadingFirstTime) {
        if (newIsLoadingFirstTime != oldIsLoadingFirstTime) {
            RecyclerView.Adapter adapter = recyclerView.getAdapter();
            if (adapter instanceof LibBaseRecyclerViewAdapter) {
                LibBaseRecyclerViewAdapter baseRecyclerViewAdapter = (LibBaseRecyclerViewAdapter) adapter;
                baseRecyclerViewAdapter.setHasProgress(newIsLoadingFirstTime);
            }
        }
    }
}

File: library/src/main/java/com/github/ykrank/androidtools/binding/LibRecyclerViewBindingAdapter.kt
package com.github.ykrank.androidtools.binding

import androidx.recyclerview.widget.RecyclerView
import com.github.ykrank.androidtools.ui.adapter.LibBaseRecyclerViewAdapter

object LibRecyclerViewBindingAdapter {
    @JvmStatic
    fun setHasProgress(
        recyclerView: RecyclerView,
        oldIsLoadingFirstTime: Boolean?,
        newIsLoadingFirstTime: Boolean?
    ) {
        if (newIsLoadingFirstTime != oldIsLoadingFirstTime) {
            val adapter = recyclerView.adapter
            if (adapter is LibBaseRecyclerViewAdapter) {
                adapter.setHasProgress(newIsLoadingFirstTime!!)
            }
        }
    }
}
